logstash-core/src/main/java/org/logstash/config/ir/compiler/DatasetCompiler.java

```diff
@@ -73,19 +76,23 @@ private DatasetCompiler() {
             );
         }
         return ComputeStepSyntaxElement.create(
-            Arrays.asList(compute.compute(), compute.clear(), MethodSyntaxElement.right(right)),
-            compute.fields(), SplitDataset.class
+            vertexId, Arrays.asList(compute.compute(), compute.clear(),
```

Comment:
nitpick: when splitting a long list of arguments, I prefer following a similar one-per-line as you do in defining functions to arbitrary newline.

Reply:
totally agree ... I tried to minimize changes and I resisted the urge to reformat here.